src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/BrokeredServices/WorkspaceProjectSystemService.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Remote.ProjectSystem;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace.BrokeredServices;
+
+#pragma warning disable RS0030 // This is intentionally using System.ComponentModel.Composition for compatibility with MEF service broker.
+/// <summary>
+/// An implementation of the brokered service <see cref="IWorkspaceProjectSystemService"/> that just maps calls to the underlying project system.
+/// </summary>
+[ExportBrokeredService("Microsoft.VisualStudio.LanguageServices.WorkspaceProjectSystemService", null, Audience = ServiceAudience.Local)]
+internal sealed class WorkspaceProjectSystemService : IWorkspaceProjectSystemService, IExportedBrokeredService
+{
+    private readonly LanguageServerProjectSystem _projectSystem;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public WorkspaceProjectSystemService(LanguageServerProjectSystem projectSystem, ProjectInitializationHandler projectInitializationHandler)
+    {
+        _projectSystem = projectSystem;
+    }
+
+    ServiceRpcDescriptor IExportedBrokeredService.Descriptor => WorkspaceProjectSystemServiceDescriptor.ServiceDescriptor;
+
+    Task IExportedBrokeredService.InitializeAsync(CancellationToken cancellationToken)
+    {
+        return Task.CompletedTask;
+    }
+
+    async Task<ILoadedProjectSystemProject> IWorkspaceProjectSystemService.LoadProjectAsync(string projectFilePath, ImmutableDictionary<string, string> buildProperties, CancellationToken cancellationToken)
```

Comment:
My expectation was that they'd only be calling one at a time since the projects are loading in parallel on their side and they won't really know the batching. But that might be different in that case; @lifengl or @tmeschter?